### PR TITLE
[ipcamera] Improve ONVIF discovery and bug fixes.

### LINKED ISSUE #9022: [ipcamera] ONVIF Discover failing due to probe type
Using 2.5.10, trying to discover an ONVIF IP Camera fails silently. This camera is discovered correctly by other ONVIF software.

Sniffing the network, the discovery message is sent but not answered by the camera.

**Discovery message sent by the binding:**

```
<?xml version="1.0" encoding="UTF-8"?>
<e:Envelope xmlns:e="http://www.w3.org/2003/05/soap-envelope" xmlns:w="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:d="http://schemas.xmlsoap.org/ws/2005/04/discovery" xmlns:dn="http://www.onvif.org/ver10/network/wsdl">
  <e:Header>
    <w:MessageID>uuid:b1bf6b3f-ab94-42f3-a225-c75e2f0b00c1</w:MessageID>
    <w:To e:mustUnderstand="true">urn:schemas-xmlsoap-org:ws:2005:04:discovery</w:To>
    <w:Action a:mustUnderstand="true">http://schemas.xmlsoap.org/ws/2005/04/discovery/Probe</w:Action>
  </e:Header>
  <e:Body>
    <d:Probe>
      <d:Types xmlns:dp0="http://www.onvif.org/ver10/network/wsdl">dp0:NetworkVideoTransmitter</d:Types>
    </d:Probe>
  </e:Body>
</e:Envelope>
```

Looking at the message sent by other software, I see that the camera does indeed respond when changing the probe type tag from:

```
<d:Types xmlns:dp0="http://www.onvif.org/ver10/network/wsdl">dp0:NetworkVideoTransmitter</d:Types>
```

To:

```
<d:Types xmlns:d="http://schemas.xmlsoap.org/ws/2005/04/discovery" xmlns:dp0="http://www.onvif.org/ver10/network/wsdl">dp0:NetworkVideoTransmitter</d:Types>
```

Or just (as seen in the ONVIF examples documentation):

```
<d:Types>dp0:NetworkVideoTransmitter</d:Types>
```

Could the probe message be changed to allow for these cameras to be discovered?

### LINKED ISSUE #9023: [ipcamera] ONVIF discovery uses only the first listed network interface
Using 2.5.10. 

I have several network interfaces on my system, and the camera happens not to be on the first one, so I can only use discovery by disabling other interfaces. I might also end up having cameras on different interfaces (wired and Wifi).

Looking into the binding code, I seem to understand the discovery only uses the first available IP interface.

Would it be possible to use the primary interface defined in Openhab configuration if defined, instead of the first listed one?

At a second stage, would it be possible to launch discovery on all available interfaces?

I would be happy to help with code, but I do not have a develpment set-up so not easy for me to compile / test / commit binding code

### LINKED ISSUE #9167: [ipcamera] Should not write password to log
## Expected Behavior
When writing to the log, passwords should be masked.

## Current Behavior
Some actions (for example recordMP4) are writing the password to the log.

## Steps to Reproduce (for Bugs)
1. Set the log level to DEBUG
2. Start action recordMP4

```
2020-11-29 12:01:44.313 [DEBUG] [ng.ipcamera.internal.IpCameraActions] - Recording 3_2020-11-29_12-01-44.mp4 for 5 seconds.
2020-11-29 12:01:44.315 [DEBUG] [hab.binding.ipcamera.internal.Ffmpeg] - Starting ffmpeg with this command now:-y -t 5 -hide_banner -loglevel warning -rtsp_transport tcp -i rtsp://admin:myPassword@192.168.0.18:554/cam/realmonitor?channel=1&amp;subtype=0&amp;unicast=true&amp;proto=Onvif -c:v copy -c:a copy /video/ipcamera/3_2020-11-29_12-01-44.mp4
```

## Your Environment
openHAB 2.5.10 and 3.0.0 M3.

### PR DESCRIPTION
**Pull request by Skinah**
Closed #9167
Closed #9023
Closed #9022

This PR changes the following:

- Fix Offline detection. Would not detect the camera was offline when not polling a snapshot.
- Improve ONVIF discovery. Now scans on all NICs instead of only the first interface the binding would find.
- Fix motion alarm FFmpeg options. Config was not working correctly if the user supplied options was '-vf xxxxxx' due to binding already providing this argument already.
- Improve log by removing password from logged FFmpeg commands.
- Refactor some code to make it slightly faster and consistent between similar classes.

### CODE REVIEW

File: bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/Ffmpeg.java [hunk 1]
```diff
@@ -169,7 +171,8 @@ public void run() {
     public void startConverting() {
         if (!ipCameraFfmpegThread.isAlive()) {
             ipCameraFfmpegThread = new IpCameraFfmpegThread();
-            logger.debug("Starting ffmpeg with this command now:{}", ffmpegCommand);
+            logger.debug("Starting ffmpeg with this command now:{}",
+                    ffmpegCommand.replaceAll(password, "passwordRemoved"));
```
Comment (cpmeister) [inline bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/Ffmpeg.java:175]:
```suggestion
                    ffmpegCommand.replaceAll(password, "********"));
```

File: bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/HikvisionHandler.java [hunk 2]
```diff
@@ -64,18 +64,13 @@ public HikvisionHandler(ThingHandler handler, int nvrChannel) {
...
-        int debounce = 3;
         try {
-            content = msg.toString();
-            if (content.isEmpty()) {
-                return;
-            }
+            int debounce = 3;
+            String content = msg.toString();
             logger.trace("HTTP Result back from camera is \t:{}:", content);
-
```
Comment (cpmeister) [inline bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/HikvisionHandler.java:78]: `.toString()` might be an expensive operation, so calling it twice when you only need to call it once is a bad practice. The way it was before only called it once. What were you trying to address here with this change?

**Review by cpmeister — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/HikvisionHandler.java (hunk 2 above)
Comment (Skinah) [reply bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/HikvisionHandler.java]: FIXED: I looked into it and what I was trying to address does not happen anymore. I have removed it back to only calling it once, thanks for pointing it out.

File: bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/onvif/OnvifDiscovery.java [hunk 3]
```diff
@@ -213,26 +215,21 @@ protected void channelRead0(@Nullable ChannelHandlerContext ctx, DatagramPacket
...
+        for (NetworkInterface nic : nics) {
+            DatagramChannel datagramChannel = (DatagramChannel) bootstrap.option(ChannelOption.IP_MULTICAST_IF, nic)
+                    .bind(new InetSocketAddress(0)).sync().channel();
+            datagramChannel
+                    .joinGroup(new InetSocketAddress(InetAddress.getByName("239.255.255.250"), 3702), networkInterface)
+                    .sync();
+            openChannels.add(datagramChannel);
+        }
+        if (!openChannels.isEmpty()) {
+            openChannels.writeAndFlush(wsDiscovery());
```
Comment (cpmeister) [inline bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/onvif/OnvifDiscovery.java:228]: Does this still return a future that you should wait for? Or do you not need to do that anymore?

Comment (Skinah) [reply bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/onvif/OnvifDiscovery.java:228]: I don’t need to use a future as it gives no benefits and only increases code length.

**Review by cpmeister — APPROVED** [review]

**Pull request merged**